src/Microsoft.Health.Functions.Extensions/AzureFunctionsJobHostBuilder.cs

```diff
@@ -78,6 +83,13 @@ public IHost Build()
                         .AddEnvironmentVariables();
                 })
             .ConfigureLogging((c, b) => _configureLogger(c, b))
+            .ConfigureServices(services =>
+            {
+                services.AddSingleton<TelemetryClient>(new TelemetryClient(new TelemetryConfiguration()
+                {
+                    TelemetryChannel = Substitute.For<ITelemetryChannel>(),
```

Comment:
Instead of using a mock, can we create a simple `NullTelemetryChannel` that just no-ops all of the calls for `ITelemetryChannel`? Then we can also add some more services users may expect (in particular the `TelemetryConfiguration`):

```csharp
.ConfigureServices(services => services
    .AddSingleton<ITelemetryChannel, NullTelemetryChannel>()
    .AddSingleton(sp => new TelemetryConfguration { TelemetryChannel = sp.GetRequiredService<ITelemetryChannel>() })
    .AddSingleton<TelemetryClient>())

```